### PR TITLE
Add ability to generate report in cobertura format

### PR DESCRIPTION
**Pull request by arnaspet**
Added cobertura format support, also updated docs.

### CODE REVIEW

File: src/CodeCoverageExtension.php [hunk 1]
```diff
@@ -139,6 +139,8 @@ public function load(ServiceContainer $container, array $params = []): void
                         $reports['html'] = new Report\Html\Facade();
 
                         break;
+                    case 'cobertura':
+                        $reports['cobertura'] = new Report\Cobertura();
```
Comment (jaylinski) [inline src/CodeCoverageExtension.php:143]: Please add a `break;` statement like above.

**Review by jaylinski — CHANGES_REQUESTED** [review]

File: src/CodeCoverageExtension.php (hunk 1 above)
Comment (arnaspet) [reply src/CodeCoverageExtension.php:143]: whoops. 😅 added.

**Review by jaylinski — APPROVED** [review]

**Comment by shulard** [comment]
Hello @arnaspet !

Thank you for the given change. I'm ok with the commit but there are some issues during PHPSpec execution. Can you take a look ? I'll merge the change when the workflow will be green.

**Comment by jaylinski** [comment]
@shulard I ran all tests on my MacBook with PHP 7.4 and couldn't reproduce those errors. Can you restart the CI pipeline? Maybe it was a fluke...

**Comment by arnaspet** [comment]
yeah, really strange error, can't reproduce.

**Comment by shulard** [comment]
Hello !

I've just relaunched the CI pipeline and the error is still here in the final result. @drupol, since you worked on the GitHub action part, can you take a look ?

**Comment by drupol** [comment]
This is an issue that I also have. I don't know what is the issue, I've already spent a couple of hours.

In the end, I disabled the MacOS tests.

**Comment by jaylinski** [comment]
@shulard I verified the failed CI tests in #50. I made no changes, but the pipelines still failed. I'd say to just merge this PR anyway (I guess you can override the CI checks). 

Maybe we have to reduce the amount of jobs we are running?

**Comment by arnaspet** [comment]
Some update: I updated my macOS and after messing a bit with xcode & php version by accident some of my own phpspec tests started failing. Tried running tests inside this repo , same happened.

so i found out that it might be related with phpspec tokenizer. especially this line
https://github.com/phpspec/phpspec/blob/30dc06a0f20642d9589bb644fbcfcb1692828699/src/PhpSpec/CodeAnalysis/TokenizedNamespaceResolver.php#L119

my test.php script looks like this
```
<?php
echo preg_match('/\\\\([^\\\\]+)\s*$/', ' Bunny\Channel', $matches) . "\n";
echo preg_match('/\\\\([^\\\\]+)\s*$/', ' App\Factories\RabbitMQFactory', $matches);
```

inside my dockerised ( working fine ) php i get
```
1
1
```

but inside broken one i get 
```
1
0
```

**Comment by drupol** [comment]
This is very interesting !

I don't have any MacOS at home nor at work so I cannot work on this; but it could be nice to make a kind of reproducer and open an issue on phpspec/phpspec.

**Comment by drupol** [comment]
@arnaspet I opened the issue on phpspec/phpspec, we'll see how it goes.

However, I asked to multiple people to run the commands you pasted and they all return `1 1`.

So, as you have something different, could you please give us some info on the PHP version your using ? Thanks!

**Comment by arnaspet** [comment]
@drupol it's 
```
➜ php --version
PHP 7.4.20 (cli) (built: Jun  4 2021 03:32:07) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Xdebug v3.0.0, Copyright (c) 2002-2020, by Derick Rethans
```

**Comment by drupol** [comment]
Could you please trigger the tests once again, it should be fixed now.

**Comment by arnaspet** [comment]
@shulard 👋

**Comment by shulard** [comment]
Hello !

I haven't the ability to rerun the workflow anymore (too old) so I just merged the code. If there are still some issues on master, please open a new PR to fix them.

**Pull request merged**